src/style-spec/expression/evaluation_context.js

```diff
@@ -47,6 +54,33 @@ class EvaluationContext {
         return this.feature && this.feature.properties || {};
     }
 
+    distanceFromCenter() {
+        if (this.featureTileCoord && this.featureDistanceData) {
+
+            const c = this.featureDistanceData.center;
+            const scale = this.featureDistanceData.scale;
+            const angle = this.featureDistanceData.angle;
+            const {x, y} = this.featureTileCoord;
+
+            // Calculate the distance vector `d`
+            const dX = (x * scale - c[0]) * this.featureDistanceData.windowScaleFactor;
+            const dY = (y * scale - c[1]) * this.featureDistanceData.windowScaleFactor;
+
+            // Calculate the bearing vector `v` by rotating unit vector clockwise
+            const b = [0, -1];
+            const cos = Math.cos(-angle);
+            const sin = Math.sin(-angle);
+            const bX = cos * b[0] - sin * b[1];
+            const bY = sin * b[0] + cos * b[1];
```

Comment:
Nice change on removing the matrix, though I would move this section in the cold path even if it adds one extra value, so that we're not calling `cos`/`sin` here and storing it in the cache. `bX`, `bY` should be the same for all symbols on screen, whereas we'd call it for each feature here.